src/MonitoringFunctions.Common/TeamsConnector.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MonitoringFunctions.Common
+{
+    /// <summary>
+    /// Represents a connector for MS Teams.
+    /// </summary>
+    internal sealed class TeamsConnector
+    {
+        private static readonly HttpClient HttpClient = new HttpClient();
+        private const string IncidentCardFormat = "{{\"@context\":\"https://schema.org/extensions\",\"@type\":\"MessageCard\",\"themeColor\":\"FF5555\",\"title\":\"{0}\",\"text\":\"{1}\",\"potentialAction\":[{{\"@type\":\"OpenUri\",\"name\":\"Go To Work Item\",\"targets\":[{{\"os\":\"default\",\"uri\":\"{2}\"}}]}}]}}";
+        
+        private string _webhookUrl;
+
+        /// <summary>
+        /// Creates a new instance of type <see cref="TeamsConnector"/>.
+        /// </summary>
+        /// <param name="webhookUrl">Webhook URL that can be used to access this connector on Teams.</param>
+        /// <exception cref="ArgumentException">If the supplied webhook url is null or whitespace.</exception>
+        public TeamsConnector(string webhookUrl)
+        {
+            if(string.IsNullOrWhiteSpace(webhookUrl))
```

Comment:
consider:
`_webhookUrl = (string.IsNullOrWhiteSpace(webhookUrl) ?  throw new ArgumentNullException(paramName: nameof(webhookUrl)) : webhookUrl;`

Reply:
Argument null imho means it is null, not empty or whitespace. The current exception is correct.

Reply:
About the styling suggestion: Though I like the one-line alternative in some cases, I think it is usually making the code harder to read. [Other examples](https://github.com/dotnet/runtime/search?p=9&q=argumentnullexception&type=code) I could find in the libraries are also styled this way. So I will keep this as is.